rust/agama-server/src/storage/web/zfcp.rs

```diff
@@ -0,0 +1,254 @@
+            post(deactivate_disk),
+        )
+        .route("/disks", get(get_disks))
+        .route("/probe", post(probe))
+        .route("/config", get(get_config))
+        .with_state(state);
+    Ok(router)
+}
+
+/// Returns whether zFCP technology is supported or not
+#[utoipa::path(
+    get,
+    path="/supported",
+    context_path="/api/storage/zfcp",
+    responses(
+        (status = OK, description = "Returns whether ZFCP technology is supported")
+    )
+)]
+async fn supported(State(state): State<ZFCPState<'_>>) -> Result<Json<bool>, Error> {
+    Ok(Json(state.client.supported().await?))
+}
+
+/// Returns global zFCP configuration
+#[utoipa::path(
+    get,
+    path="/config",
+    context_path="/api/storage/zfcp",
+    responses(
+        (status = OK, description = "Returns global ZFCP configuration", body=ZFCPOptions)
+    )
+)]
+async fn get_config(State(state): State<ZFCPState<'_>>) -> Result<Json<ZFCPOptions>, Error> {
+    Ok(Json(ZFCPOptions {
+        allow_lun_scan: state.client.is_lun_scan_allowed().await?,
+    }))
+}
+
+/// Returns the list of known zFCP disks.
+#[utoipa::path(
+    get,
+    path="/disks",
+    context_path="/api/storage/zfcp",
+    responses(
+        (status = OK, description = "List of ZFCP disks", body = Vec<ZFCPDisk>)
+    )
+)]
+async fn get_disks(State(state): State<ZFCPState<'_>>) -> Result<Json<Vec<ZFCPDisk>>, Error> {
+    let devices = state
+        .client
+        .get_disks()
+        .await?
+        .into_iter()
+        .map(|(_path, device)| device)
+        .collect();
+    Ok(Json(devices))
+}
+
+/// Returns the list of known zFCP controllers.
+#[utoipa::path(
+    get,
+    path="/controllers",
+    context_path="/api/storage/zfcp",
+    responses(
+        (status = OK, description = "List of ZFCP controllers", body = Vec<ZFCPController>)
+    )
+)]
+async fn controllers(
+    State(state): State<ZFCPState<'_>>,
+) -> Result<Json<Vec<ZFCPController>>, Error> {
+    let devices = state
+        .client
+        .get_controllers()
+        .await?
+        .into_iter()
+        .map(|(_path, device)| device)
+        .collect();
+    Ok(Json(devices))
+}
+
+/// Activate given zFCP controller.
+#[utoipa::path(
+    post,
+    path="/controllers/:controller_id/activate",
+    context_path="/api/storage/zfcp",
+    responses(
+        (status = OK, description = "controller activated")
+    )
+)]
+async fn activate_controller(
+    State(state): State<ZFCPState<'_>>,
+    Path(controller_id): Path<String>,
+) -> Result<Json<()>, Error> {
+    state
+        .client
+        .activate_controller(controller_id.as_str())
+        .await?;
+    Ok(Json(()))
+}
+
+/// List wwpns for given controller.
+#[utoipa::path(
+    post,
+    path="/controllers/:controller_id/wwpns",
+    context_path="/api/storage/zfcp",
+    responses(
+        (status = OK, description = "list of wwpns", body=Vec<String>)
```

Comment:
```suggestion
        (status = OK, description = "List of WWPNs", body=Vec<String>)
```

It goes to the OpenAPI docs, so please use *WWPNs* or *World Wide Port Names*. The same for the rest of endpoints.